xmr-btc/src/bitcoin/timelocks.rs

```diff
@@ -0,0 +1,48 @@
+use serde::{Deserialize, Serialize};
+use std::ops::Add;
+
+/// Represent a timelock, expressed in relative block height.
+/// E.g. The timelock expires 10 blocks after the reference transaction is
+/// mined.
+#[derive(Debug, Copy, Clone, Serialize, Deserialize, Eq, PartialEq)]
+#[serde(transparent)]
+pub struct Timelock(u32);
+
+impl Timelock {
+    pub const fn new(relative_block_timelock: u32) -> Self {
```

Comment:
🙃 For me it would be more logical to just name the parameter `nr_of_blocks` as this is what the `u32` parameter specifies.